community/bolt/src/main/java/org/neo4j/bolt/v1/runtime/BoltStateMachine.java

```diff
@@ -290,14 +296,20 @@ public String owner()
     @Override
     public void terminate()
     {
-        close();
+        /*
+         * This is a side-channel call and we should not close anything directly.
+         * Just mark the transaction and set isTerminated to true and then the session
+         * thread will close down the connection eventually.
+         */
+        ctx.isTerminated.set( true );
+        ctx.statementProcessor.markCurrentTransactionForTermination();
         spi.onTerminate( this );
     }
 
     @Override
-    public boolean hasTerminated()
+    public boolean willTerminate()
     {
-        return isClosed();
```

Comment:
Does this warrant removing `isClosed()`? I notice it's `public`, but it's not part of an interface and only used by one test matcher, so perhaps?